frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -21,275 +21,351 @@
+			let input = mtf::encode_cross_receive(
+				lock_info.mapped_address,
+				lock_info.recipient,
+				lock_info.amount,
+			)
+			.map_err(|_| Error::<T>::InvalidEncodeERC20)?;
 			Self::transact_mapping_factory(input)?;
-			VerifiedIssuingProof::insert(tx_index, true);
-			Self::deposit_event(RawEvent::RedeemErc20(user, backing_address, tx_index));
+			VerifiedIssuingProof::<T>::insert(tx_index, true);
+			Self::deposit_event(Event::RedeemErc20(backing_address, tx_index));
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::mapping_factory_event_handle())]
+		pub fn mapping_factory_event_handle(
+			origin: OriginFor<T>,
+			input: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let caller = ensure_signed(origin)?;
+			ensure!(input.len() >= 8, <Error<T>>::InvalidInput);
+			let factory = MappingFactoryAddress::<T>::get();
+			let factory_id = <T as darwinia_evm::Config>::AddressMapping::into_account_id(factory);
+			ensure!(factory_id == caller, <Error<T>>::AssetAR);
+			let burn_action = &sha3::Keccak256::digest(&BURN_ACTION)[0..4];
+			let register_action = &sha3::Keccak256::digest(&REGISTER_ACTION)[0..4];
+			if &input[4..8] == burn_action {
+				let burn_info =
+					TokenBurnInfo::decode(&input[8..]).map_err(|_| Error::<T>::InvalidInputData)?;
+				ensure!(burn_info.recipient.len() == 20, <Error<T>>::AssetAR);
+				Self::deposit_burn_token_event(
+					burn_info.backing,
+					burn_info.sender,
+					burn_info.source,
+					EthereumAddress::from_slice(burn_info.recipient.as_slice()),
+					burn_info.amount,
+				)?;
+			} else if &input[4..8] == register_action {
+				let register_info = TokenRegisterInfo::decode(&input[8..])
+					.map_err(|_| Error::<T>::InvalidInputData)?;
+				Self::finish_token_registered(register_info.0, register_info.1, register_info.2);
+			} else {
+				log::trace!("Unsupport action!");
+			}
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::set_mapping_factory_address())]
+		pub fn set_mapping_factory_address(
+			origin: OriginFor<T>,
+			address: EthereumAddress,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let old_address = MappingFactoryAddress::<T>::get();
+			MappingFactoryAddress::<T>::put(address);
+			Self::deposit_event(Event::MappingFactoryAddressUpdated(old_address, address));
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::set_ethereum_backing_address())]
+		pub fn set_ethereum_backing_address(
+			origin: OriginFor<T>,
+			address: EthereumAddress,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let old_address = EthereumBackingAddress::<T>::get();
+			EthereumBackingAddress::<T>::put(address);
+			Self::deposit_event(Event::EthereumBackingAddressUpdated(old_address, address));
+			Ok(().into())
 		}
 	}
 }
 
-impl<T: Config> Module<T> {
+impl<T: Config> Pallet<T> {
 	/// The account ID of the issuing pot.
-	pub fn dvm_account_id() -> H160 {
+	pub fn dvm_account_id() -> EthereumAddress {
```

Comment:
Is this function no longer in use? Compile won't fire when I remove it.

Reply:
resolved. https://github.com/darwinia-network/darwinia-common/compare/b23ee2338082...864e1726f093